app/assets/javascripts/admin/order_cycles/controllers/incoming_controller.js.coffee

```diff
@@ -1,7 +1,27 @@
-angular.module('admin.orderCycles').controller 'AdminOrderCycleIncomingCtrl', ($scope, $controller, $location, Enterprise, ocInstance) ->
+angular.module('admin.orderCycles').controller 'AdminOrderCycleIncomingCtrl', ($scope, $controller, $location, Enterprise, OrderCycle, ocInstance) ->
   $controller('AdminOrderCycleExchangesCtrl', {$scope: $scope, ocInstance: ocInstance, $location: $location})
 
   $scope.view = 'incoming'
 
-  $scope.enterpriseTotalVariants = (enterprise) ->
-    Enterprise.totalVariants(enterprise)
+  $scope.exchangeTotalVariants = (exchange) ->
+    return unless this.enterprises? && this.enterprises[exchange.enterprise_id]?
```

Comment:
I could be totally wrong here, but I don't think referencing `this` is normally used in Angular. In the context of a controller I think `$scope` is essentially `this`?

Reply:
they are not exactly equivalent.
at some point $scope was not working and this was. there's a hirerchy of scopes and controllers here... maybe that was causing the trouble.
anyway, fixed in new commit. 26d3219deec3ecac061190f0db1a75774056af9f
thanks :+1: